pkg/controller/datavolume/smart-clone-controller.go

```diff
@@ -356,9 +356,22 @@ func (r *SmartCloneReconciler) getTargetPVC(dataVolume *cdiv1.DataVolume) (*core
 func newPvcFromSnapshot(dv *cdiv1.DataVolume, name string, snapshot *snapshotv1.VolumeSnapshot, targetPvcSpec *corev1.PersistentVolumeClaimSpec) (*corev1.PersistentVolumeClaim, error) {
 	targetPvcSpecCopy := targetPvcSpec.DeepCopy()
 	restoreSize := snapshot.Status.RestoreSize
-	if restoreSize == nil {
+	if restoreSize == nil || restoreSize.Sign() == -1 {
```

Comment:
Are we sure we want to error out on the negative sign?
I mean, if the volumesnapshot is ready, and for some reason, the provisioner reports a negative sign, couldn't we proceed in the same way as if it was zero?

Just want to make sure we won't be back with the same issue we have with `0`

Reply:
Same for nil `.Status.RestoreSize`, should we be able to work with that as well?

Reply:
I was originally going to allow missing restoreSize, but if it is missing then that means there is a serious problem with the snapshot, and we should error out, same with a negative number.